crates/ruff/src/checkers/physical_lines.rs

```diff
@@ -154,6 +160,24 @@ pub(crate) fn check_physical_lines(
                 diagnostics.push(diagnostic);
             }
         }
+        if enforce_copyright_header && !checked_copyright_header_absent {
+            let missing_copyright_header =
+                copyright_header_absent(&line, settings, chars_before_copyright_header);
+
+            match missing_copyright_header {
+                Some(false) => checked_copyright_header_absent = true,
+                Some(true) => {
+                    let diagnostic = Diagnostic::new(
+                        HeaderLacksCopyright,
+                        TextRange::new(line.start(), line.end()),
+                    );
+                    diagnostics.push(diagnostic);
+                    checked_copyright_header_absent = true;
+                }
+                None => (),
+            }
+            chars_before_copyright_header += line.len() as i64;
```

Comment:
`line.len` gives you the number of bytes on that line, not the number of characters. The two numbers are different for non-ascii strings (e.g. emojis consist of multiple bytes, but represent a single character)